source/btle/btle.cpp

```diff
@@ -183,6 +183,7 @@ static void btle_handler(ble_evt_t *p_ble_evt)
                     break;
             }
             nRF5xGap::getInstance().processDisconnectionEvent(handle, reason);
+            // TODO: close pending discoveries
```

Comment:
I have made the changes. I have added a new function in `nRF5xServiceDiscovery` : `void terminate(Gap::Handle_t connectionHandle)`. It is an overload of terminate, if the handle passed in parameters match the handle of the current discovery, it terminate the discovery otherwise, the discovery continue to run.